### PR TITLE
ContactDetailsFormFields: add optional error message prop

### PR DESCRIPTION
**Pull request by nbloomf**
#### Changes proposed in this Pull Request

* In composite-checkout we are using `ContactDetailsFormFields` as a managed component. This patch adds an optional prop, `contactDetailsErrors`, which can be used to pass in per-field errors through props. These errors have lower precedence than those triggered by the existing validation logic.

#### Testing instructions

* On its own this PR should have no visible effects on the contact details form. Verify that this is the case by interacting with the form in checkout. **Since this touches a shared component you'll need to verify this behavior in both composite checkout and old checkout.**

### CODE REVIEW

File: client/components/domains/contact-details-form-fields/custom-form-fieldsets/eu-address-fieldset.jsx [hunk 1]
```diff
@@ -17,13 +17,13 @@ const EuAddressFieldset = props => {
 		<div className="custom-form-fieldsets__address-fields eu-address-fieldset">
 			<Input
 				label={ translate( 'Postal Code' ) }
-				errorMessage={ contactDetailsErrors?.postalCode }
 				{ ...getFieldProps( 'postal-code' ) }
+				errorMessage={ contactDetailsErrors?.postalCode }
```
Comment (sirbrillig) [inline client/components/domains/contact-details-form-fields/custom-form-fieldsets/eu-address-fieldset.jsx]: Does `getFieldProps()` return `errorMessage`? If so, will this override it even if it's `undefined`? Or does React allow that to work?

**Comment by sirbrillig** [comment]
I pushed an alternate way of including the error message. Let me know what you think.

**Comment by sirbrillig** [comment]
Hm... this breaks existing checkout error displays.

Screen Shot 2020-03-06 at 9 18 35 PM

**Comment by sirbrillig** [comment]
I figured it out. I forgot about `createField`.

**Comment by sirbrillig** [comment]
Tested with #39842 and fixed a mistake I made.

**Comment by nbloomf** [comment]
Your updates look good and contact details in regular checkout works as expected. Thank you for having a look at this!

**Pull request merged**